src/renderer/drawer.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from "blockly/core";
+
+export class Drawer extends Blockly.zelos.Drawer {
+  drawStatementInput_(row) {
+    if (this.info_.isBowlerHatBlock()) {
+      // Bowler hat blocks have straight sides with no C-shape/indentation for
+      // statement blocks.
+      this.drawRightSideRow_(row);
+      this.positionStatementInputConnection_(row);
+    } else {
+      super.drawStatementInput_(row);
+    }
+  }
+
+  drawRightSideRow_(row) {
+    if (
+      this.info_.isBowlerHatBlock() &&
+      Blockly.blockRendering.Types.isSpacer(row)
+    ) {
+      Blockly.blockRendering.Drawer.prototype.drawRightSideRow_.call(this, row);
+    } else {
+      super.drawRightSideRow_(row);
+    }
+  }
+
+  drawTop_() {
+    super.drawTop_();
+    // This is a horrible hack, but the superclass' implementation of drawTop_()
+    // provides no way to cleanly override a hat's path without copying and
+    // pasting the entire implementation here. We know that there will only be
+    // one hat on a block, and its path is a known constant, so we just find and
+    // replace it with the desired bowler hat path here.
```

Comment:
😢

but correct!

You can also link this to https://github.com/google/blockly/issues/7292 if it makes you feel better.